src/main.rs

```diff
@@ -29,12 +39,32 @@ async fn main() {
     let ws_rpc_url = std::env::var("WS_RPC_URL").expect("WS_RPC_URL not set");
     let rpc_url = std::env::var("RPC_URL").expect("RPC_URL not set");
     let mnemonic = std::env::var("NFT_MINTER_MNEMONIC").expect("NFT_MINTER_MNEMONIC not set");
-    let tls_key_path = std::env::var("TLS_KEY_PATH").expect("TLS_KEY_PATH not set");
-    let tls_cert_path = std::env::var("TLS_CERT_PATH").expect("TLS_CERT_PATH not set");
     let db_path = std::env::var("DB_PATH").expect("DB_PATH not set");
     let app_url = std::env::var("APP_URL").expect("APP_URL not set");
     let tee_url = std::env::var("TEE_URL").expect("TEE_URL not set");
 
+    let pkey = if std::path::Path::new(PRIVATE_KEY_PATH).exists() {
+        let pk_bytes = fs::read(PRIVATE_KEY_PATH).await.expect("Failed to read pk file");
+        PKey::private_key_from_pem(pk_bytes.as_slice()).unwrap()
+    } else {
+        let pk = create_rsa_key(2048);
+        let pk_bytes = pk.private_key_to_pem_pkcs8().unwrap();
+        fs::write(PRIVATE_KEY_PATH, pk_bytes).await.expect("Failed to write pk to file");
+        pk
+    };
+
+    let csr = create_csr(&tee_url, &pkey).unwrap();
+    let csr_pem_bytes = csr.to_pem().unwrap();
+    fs::write(CSR_PATH, csr_pem_bytes).await.expect("Failed to write csr to file");
+
+    let mut pk_bytes = pkey.public_key_to_pem().unwrap();
+    let mut csr_pem_bytes = csr.to_pem().unwrap();
+    pk_bytes.append(&mut csr_pem_bytes);
+    if let Ok(quote) = sgx_attest::sgx_attest(pk_bytes) {
+        // handle quote
+        log::info!("quote: {:?}", quote);
```

Comment:
prefer this to output to `untrustedhost/quote.dat`

Reply:
added